src/cpp/hu.bme.mit.inf.modes3.ESP32-sensor/railsensor_3/ESP32-sensor/lib/SensorMonitor/sensor_monitor.cpp

```diff
@@ -0,0 +1,230 @@
+
+#include "sensor_monitor.h"
+
+
+float convertRawGyro(int gRaw) {
+  // since we are using 250 degrees/seconds range
+  // -250 maps to a raw value of -32768
+  // +250 maps to a raw value of 32767
+
+  float g = (gRaw * 250.0) / 32768.0;
+
+  return g;
+}
+
+
+void updateCallback(void *arg) {
```

Comment:
@hegyibalint expertise required

Reply:
This is a very ugly solution you applied here. A more sustainable approach in the FreeRTOS system would be:
- In the main loop, a thread is created
- The started thread create the SerialMonitor object
- The thread will loop, and use this SerialMonitor

A serious issue which might not happen here, but it's good to plan against is the fact the object passed to the thread is not owned by it (because it was passed as a pointer), moreover it's not allocated on its stack. Conclusively, one can access, and for example, call the destructor of the SerialMonitor class.